more_itertools/more.py

```diff
@@ -3683,3 +3686,77 @@ def value_chain(*args):
             yield from value
         except TypeError:
             yield value
+
+
+def product_index(element, *args):
+    """Equivalent to ``list(product(*args)).index(element)``
+
+    The products of *args* can be ordered lexicographically.
```

Comment:
Most docstrings in this library have a clear example.  I've notice these (and the *_nth combinatorics actually) are lacking them.  Example:

```python
>>> list(product(1, "abc", [1, 2, 3]))
?
```

Please include them to maintain consistency and clarity.